pkg/networkservice/chains/nsmgr/suite_test.go

```diff
@@ -394,42 +397,56 @@ func (s *nsmgrSuite) Test_PassThroughRemoteUsecase() {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
 
+	labels := []string{"a", "b", "c", "d", "e", "f", "g"}
+
+	nsReg := linearNS(labels)
+	nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, nsReg)
+	require.NoError(t, err)
+
 	var nsesReg [nodesCount]*registry.NetworkServiceEndpoint
 	for i := 0; i < nodesCount; i++ {
+		nsesReg[i] = &registry.NetworkServiceEndpoint{
+			Name:                fmt.Sprintf("endpoint-%v", labels[i]),
+			NetworkServiceNames: []string{nsReg.GetName()},
+			NetworkServiceLabels: map[string]*registry.NetworkServiceLabels{
+				nsReg.GetName(): {
+					Labels: map[string]string{
+						step: labels[i],
+					},
+				},
+			},
+		}
+
 		var additionalFunctionality []networkservice.NetworkServiceServer
-		if i != 0 {
-			// Passtrough to the node i-1
+		if i != nodesCount-1 {
 			additionalFunctionality = []networkservice.NetworkServiceServer{
 				chain.NewNetworkServiceServer(
```

Comment:
Seems like all test methods use the same code for generating the endpoint.
I suggest extracting a separate function for this.

Reply:
it is exactly the same only in 2 tests, i think function is not necessary

Reply:
You have 3 passthrough test functions, and all of which create their servers the same way.
Even if we don't add more tests, this is too much duplicated code in my opinion.

Reply:
I agree, you can add some `passthroughFunctionality(...)` method to be used in all passthrough cases.